Java/Java.sublime-syntax

```diff
@@ -400,130 +346,175 @@ contexts:
           pop: true
-        - include: code
-    - match: (?=\S)
-      pop: true
-
-  object-types-inherited:
-    - match: '\b{{qualified_id}}\b(?=\s*<)'
-      scope: entity.other.inherited-class.java
+  object-types:
+    - match: '\b({{qualified_id}})(<)(?={{primitives}})'
+      captures:
+        1: support.class.java
+        2: punctuation.accessor.dot.java
+        3: punctuation.definition.generic.begin.java
+      push: object-type-generic
+    # This will match the < operator if there is no space in between.
+    - match: '\b({{qualified_id}})(<)(?!\s*(?:\d+|<|[a-z]))'
+      comment: Don't match foo<2, foo<bar, foo<<12
+      captures:
+        1: support.class.java
+        2: punctuation.accessor.dot.java
+        3: punctuation.definition.generic.begin.java
+      push: object-type-generic
+    - match: '\b({{qualified_id}})(?=\[)'
+      captures:
+        1: support.class.java
+        2: punctuation.accessor.dot.java
       push:
-        - match: '<'
-          scope: punctuation.definition.generic.begin.java
-          set:
-            - meta_scope: meta.generic.java
-            - match: '>'
-              scope: punctuation.definition.generic.end.java
+        - match: \[
+          scope: keyword.operator.array.java
+          push:
+            - meta_scope: meta.brackets.java
+            - match: \]
+              scope: keyword.operator.array.java
               pop: true
-            - include: generic-instantiation
-    - match: '\b{{qualified_id}}\b'
-      scope: entity.other.inherited-class.java
-
-  method-parameters:
-    - match: \bfinal\b
-      scope: storage.modifier.java
-    - match: '{{primitives}}'
-      scope: storage.type.primitive.java
-      set: method-parameter-after
-    - match: '{{qualified_id}}'
+            - include: code
+        - include: any_POP
+    - include: simple-object-type
+  simple-object-type:
+    - match: '\b(?:[a-zA-Z]\w*(\.))*(?:(?=UUID\b|URI\b)|(?![A-Z0-9_]{2,}\b))[A-Z]\w*\b'
       scope: support.class.java
-      set: method-parameter-after
-    - match: ','
-      scope: punctuation.separator.java
-    - match: \)
-      scope: meta.method.parameters.java punctuation.section.parens.end.java
-      pop: true
-
-  method-parameter-after:
-    - meta_content_scope: meta.method.parameters.java
-    - match: '<'
-      scope: punctuation.definition.generic.begin.java
+      captures:
+        1: punctuation.accessor.dot.java
+  object-types-inherited:
+    - match: '\b({{qualified_id}})(?=<)'
+      captures:
+        1: entity.other.inherited-class.java
+        2: punctuation.accessor.dot.java
       push:
         - meta_scope: meta.generic.java
-        - match: '>'
-          scope: punctuation.definition.generic.end.java
-          pop: true
-        - include: generic-instantiation
-    - match: '\.\.\.'
-      scope: keyword.operator.varargs.java
-    - match: '(\[)\s*(\])'
+        - match: <
+          scope: punctuation.definition.generic.begin.java
```

Comment:
Interesting 👍 it's punctuation.section.generic.begin in C++, so, either C++ is wrong and should be updated, or the style guide is wrong.